src/components/form/form.component.js

```diff
@@ -0,0 +1,511 @@
+
+    while (element) {
+      offsetTop += element.offsetTop;
+      element = element.offsetParent;
+    }
+
+    const formHeight = (offsetTop + this._form.offsetHeight) - this._window.pageYOffset;
+
+    if (!this.state.stickyFooter && formHeight > this._window.innerHeight) {
+      this.setState({ stickyFooter: true });
+    } else if (this.state.stickyFooter && formHeight < this._window.innerHeight) {
+      this.setState({ stickyFooter: false });
+    }
+  }
+
+  addUnsavedWarningListener = () => {
+    this._window.addEventListener('beforeunload', this.checkIsFormDirty);
+  }
+
+  removeUnsavedWarningListener = () => {
+    this._window.removeEventListener('beforeunload', this.checkIsFormDirty);
+  }
+
+  // This must return undefined for IE and Safari if we don't want a warning
+  /* eslint-disable consistent-return */
+  checkIsFormDirty = (ev) => {
+    if (this.state.isDirty) {
+      // Confirmation message is usually overridden by browsers with a similar message
+      const confirmationMessage = I18n.t('form.save_prompt',
+        { defaultValue: 'Do you want to leave this page? Changes that you made may not be saved.' });
+      ev.returnValue = confirmationMessage; // Gecko + IE
+      return confirmationMessage; // Gecko + Webkit, Safari, Chrome etc.
+    }
+  }
+  /* eslint-enable consistent-return */
+
+  /**
+   * stores the document - allows us to override it different contexts, such as
+   * when running tests.
+   */
+  _document = Browser.getDocument();
+
+  /**
+   * stores the window - allows us to override it different contexts, such as
+   * when running tests.
+   */
+  _window = Browser.getWindow();
+
+  /**  Handles submit and runs validation. */
+  handleOnSubmit = async (ev) => {
+    ev.preventDefault();
+
+    if (this.props.autoDisable) {
+      this.setState({ submitted: true });
+    }
+
+    if (this.props.beforeFormValidation) {
+      this.props.beforeFormValidation(ev);
+    }
+
+    const valid = await this.props.validate();
+
+    if (this.props.afterFormValidation) {
+      this.props.afterFormValidation(ev, valid, this.enableForm);
+    }
+
+    if (valid) {
+      this.resetIsDirty();
+      this.triggerSubmit(ev, valid);
+    } else {
+      this.setState({ submitted: false });
+    }
+  }
+
+  triggerSubmit(ev, valid) {
+    if (this.props.onSubmit) {
+      this.props.onSubmit(ev, valid, this.enableForm);
+    } else {
+      this._form.submit();
+    }
+  }
+
+  /** enables a form which has been disabled after being submitted. */
+  enableForm = () => {
+    this.setState({ submitted: false });
+  }
+
+  /**
+   * Serializes the inputs in the form ready for submission via AJAX
+   * https://www.npmjs.com/package/form-serialize
+   */
+  serialize = (opts) => {
+    return Serialize(this._form, opts);
+  }
+
+ /** Separates and returns HTML specific props */
+ htmlProps = () => {
+   const { onSubmit, fixedBottom, ...props } = validProps(this);
+   return props;
+ }
```

Comment:
This `htmlProps()` block of code is only indented by 1 space (should be 2 spaces).

The same indentation issue exists for `cancelButton()`, `saveButtonWithSummary()`, `formFooter()`, `orderFormButtons()`, `renderChildren()` and `render()`.

(I'm surprised this wasn't picked up by [ESLint's `indent` rule](https://eslint.org/docs/rules/indent) :thinking:)

Reply:
Yeah that is a strange one, I'll correct the indentation

Reply:
Thanks for fixing the indentation on those functions :white_check_mark:

Looking at the `.eslintrc` file, it extends `./node_modules/carbon-factory/.eslintrc`, which does configure the `indent` rule:
```
"indent": ["error", 2, {
  "VariableDeclarator": 2,
  "SwitchCase": 1
}],
```
So I wonder why it's not working here? :thinking: I wonder where else it's not working? :worried:

Reply:
Yeah it will need some digging I suspect. In this case it seemed to be relating to how some preceding functions were aligned as it complained when i corrected the indentation until all the erroneous functions were aligned correctly 🤕

Reply:
@edleeks87: Many thanks for creating FE-1870 to investigate this :thumbsup: